statsmodels/base/tests/test_penalized.py

```diff
@@ -0,0 +1,728 @@
+
+        cls.atol = 5e-3
+
+    def test_cov_type(self):
+        res1 = self.res1
+        res2 = self.res2
+
+        assert_equal(self.res1.cov_type, 'HC0')
+        cov_kwds = {'description': 'Standard Errors are heteroscedasticity '
+                    'robust (HC0)',
+                    'adjust_df': False, 'use_t': False, 'scaling_factor': None}
+        assert_equal(self.res1.cov_kwds, cov_kwds)
+        # numbers are regression test using bfgs
+        params = np.array([0.96817787574701109, 0.43674374940137434,
+                           0.33096260487556745, 0.27415680046693747])
+        bse = np.array([0.028126650444581985, 0.033099984564283147,
+                        0.033184585514904545, 0.034282504130503301])
+        assert_allclose(res2.params[:self.k_nonzero], params, atol=1e-5)
+        assert_allclose(res2.bse[:self.k_nonzero], bse, rtol=1e-6)
+        assert_allclose(res1.params[:self.k_nonzero], params, atol=self.atol)
+        assert_allclose(res1.bse[:self.k_nonzero], bse, rtol=0.02)
+
+
+class TestPenalizedGLMPoissonOracleHC(CheckPenalizedPoisson):
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        cov_type = 'HC0'
+        modp = GLM(y, x[:, :cls.k_nonzero], family=family.Poisson())
+        cls.res2 = modp.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                            disp=0)
+
+        mod = GLMPenalized(y, x, family=family.Poisson(), penal=cls.penalty)
+        mod.pen_weight *= 1.5  # same as ddiscrete Poisson
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                           disp=0)
+
+        cls.exog_index = slice(None, cls.k_nonzero, None)
+
+        cls.atol = 5e-3
+
+
+class TestPenalizedPoissonGLMOracleHC(CheckPenalizedPoisson):
+    # compare discrete Poisson and GLM-Poisson
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        cov_type = 'HC0'
+        modp = PoissonPenalized(y, x, penal=cls.penalty)
+        modp.pen_weight *= 1.5  # increased from ddiscrete Poisson 1.5
+        modp.penal.tau = 0.05
+        cls.res2 = modp.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                            disp=0)
+
+        mod = GLMPenalized(y, x, family=family.Poisson(), penal=cls.penalty)
+        mod.pen_weight *= 1.5  # increased from ddiscrete Poisson 1.5
+        mod.penal.tau = 0.05
+        cls.res1 = mod.fit(cov_type=cov_type, method='bfgs', maxiter=100,
+                           disp=0)
+
+        cls.exog_index = slice(None, None, None)
+
+        cls.atol = 1e-4
+
+
+class TestPenalizedPoissonOraclePenalized(CheckPenalizedPoisson):
+
+    @classmethod
+    def _initialize(cls):
+        y, x = cls.y, cls.x
+        modp = PoissonPenalized(y, x[:, :cls.k_nonzero], penal=cls.penalty)
+        cls.res2 = modp.fit(method='bfgs', maxiter=100, disp=0)
+
+        mod = PoissonPenalized(y, x, penal=cls.penalty)
+        #mod.pen_weight *= 1.5
```

Comment:
pls delete commented-out

Reply:
I leave it as reminder that we use a different pen_weight now